tests/test_pilot/test_pilot.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# pylint: disable=unused-argument, no-value-for-parameter
+import time
+
+from   unittest import mock
+from   unittest import TestCase
+
+import radical.pilot as rp
+
+
+# ------------------------------------------------------------------------------
+#
+class TestPilot(TestCase):
+
+    @mock.patch.object(rp.PilotManager, '__init__', return_value=None)
+    def test_pilot_uid(self, mocked_init):
+
+        pmgr = rp.PilotManager(session=None)
+        pmgr._uids    = list()
+        pmgr._uid     = 'pmgr.0000'
+        pmgr._log     = mock.Mock()
+        pmgr._prof    = mock.Mock()
+        pmgr._session = mock.Mock()
+        pmgr._session.uid = str(time.time())  # restart uid counter
+        sandbox_url = mock.Mock()
+        sandbox_url.path = './'
+        pmgr._session._get_jsurl = mock.Mock(return_value=('ssh', sandbox_url))
+        pmgr._session._get_resource_sandbox = mock.Mock(return_value=sandbox_url)
+        pmgr._session._get_session_sandbox = mock.Mock(return_value=sandbox_url)
+        pmgr._session._get_pilot_sandbox = mock.Mock(return_value=sandbox_url)
+        pmgr._session._get_client_sandbox = mock.Mock(return_value=sandbox_url)
```

Comment:
When the pilot is initialized, it calls these methods from the session. Since the session is mocked, the methods are not defined and cannot be called. As a result, the test will fail for every method call with an error from `mock`.

Mocking the method calls will allow the test to not fail, but also allows us to provide different type of `return_values`, via a `side_effect` based on what we want to test.